virtcontainers/sandbox.go

```diff
@@ -1873,8 +1873,8 @@ func (s *Sandbox) calculateSandboxMemory() int64 {
 func (s *Sandbox) calculateSandboxCPUs() uint32 {
 	mCPU := uint32(0)
 
-	for _, c := range s.config.Containers {
-		if cpu := c.Resources.CPU; cpu != nil {
+	for _, c := range s.containers {
```

Comment:
Here I think you can't simply using sandbox containers, since when creating a container in sandbox, the s.containers is updated before calculating resources, thus you could lost accounting the created container's resources. 
IMO the root cause is that the container's ContainerConfig pointer doesn't point to the instance  located in sandbox's SandboxConfig's  map of ContainerConfig, which caused the configure's not sync.

Reply:
Yes, I think there are issues  for both of the cpu and memory resource updating.

Reply:
@lifupan Yes, the root cause is that the container's ContainerConfig pointer doesn't point to the instance located in sandbox's SandboxConfig's map of ContainerConfig.

If we want to fix this way, we need to modify code in fetchSandbox->fetchContainers->newContainer, pass by reference, and more tests. 

Or we can just update sandbox's SandboxConfig's map of ContainerConfig at container.go:update(), which need a add method such like 
func (s *Sandbox) **updateConfig**() error

Which way do you perfer?

Reply:
> @lifupan Yes, the root cause is that the container's ContainerConfig pointer doesn't point to the instance located in sandbox's SandboxConfig's map of ContainerConfig.
> 
> If we want to fix this way, we need to modify code in fetchSandbox->fetchContainers->newContainer, pass by reference, and more tests.
>
I'd prefer this one, thus we can keep only one instance of container's config in memory and changed it at one place would be saw in other place.

Reply:
OK，I'll try to code it.